src/lifecycle/window.rs

```diff
@@ -102,7 +102,7 @@ impl Window {
             .with_title(title)
             .with_dimensions(user_size.into());
         if let Some(path) = settings.icon_path {
-            window = window.with_window_icon(Some(Icon::from_path(path)?));
+            window = window.with_window_icon(Some(Icon::from_path(path).unwrap()));
```

Comment:
Why change the error propagation to an unwrap?

Reply:
That's a really good question.

The unchanged code triggers this error, and I am not sure how to resolve it.

```
error[E0277]: `?` couldn't convert the error to `error::QuicksilverError`
   --> src/lifecycle/window.rs:105:72
    |
105 |             window = window.with_window_icon(Some(Icon::from_path(path)?));
    |                                                                        ^ the trait `std::convert::From<image::image::ImageError>` is not implemented for `error::QuicksilverError`
    |
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = help: the following implementations were found:
              <error::QuicksilverError as std::convert::From<gilrs::gamepad::Error>>
              <error::QuicksilverError as std::convert::From<glutin::ContextError>>
              <error::QuicksilverError as std::convert::From<glutin::CreationError>>
              <error::QuicksilverError as std::convert::From<graphics::atlas::AtlasError>>
            and 7 others
    = note: required by `std::convert::From::from`
```

I did spend some time looking at the definition at https://github.com/ryanisaacg/quicksilver/blob/master/src/error.rs#L120-L125 but I am not sure how to fix the above error.

Reply:
I think the problem is that the version of the `image` crate that Quicksilver depends on and the version of `image` that glutin depends on don't match after your PR, and therefore the types are not the same. If you make sure that they match, this error should go away.

Reply:
You're right about that.

I dropped the update of image crate from this PR and it seems to work correctly now.